### PR TITLE
Monkey + Dog AI Updates [TG Ports]

### PR DESCRIPTION
**Pull request by DoctorSquishy**
# About The Pull Request
Fixes Monkey's grabbing lights, they burn their wittle hands again
Updates Monkey and Dog AI code closer to current TG, including hard deletes and some runtimes.
Breaks old pre-subtree into each action and sorts them
Monkeys play the donkey kong theme on any instruments now, so go make a monkey band!
Dogs AI no longer hangs up when fetching objects and can growl now at targets

## Why It's Good For The Game
Makes the AI code a bit more flexible

## Testing Photographs and Procedure
Beat up some Monkeys, shove them into so DNA scanners, told Ian to growl at them, monkey's rebelled and beat me up and threw me into a nearby disposals.

## PR References
- https://github.com/tgstation/tgstation/pull/61455
- https://github.com/tgstation/tgstation/pull/57149
- https://github.com/tgstation/tgstation/pull/61741
- https://github.com/tgstation/tgstation/pull/61726
- https://github.com/tgstation/tgstation/pull/57521
- https://github.com/tgstation/tgstation/pull/65384

## Changelog
:cl: DoctorSquishy, Ryll/Shaps, Ghommie, tralezab, ghost
add: Hand a monkey an instrument. It'll play a cool song!
fix: Dog AI's should no longer lock up and become unresponsive after failed fetch attempts
soundadd: Dogs in harass mode that are guarding against someone will now make growling sounds
fix: Monkey's burn their hands on bulbs again
fix: Fixes a bunch of hard deletes + runtimes in AI Code
fix: Monkeys no longer rip apart machines from the inside 
refactor: Refactored datum AI idle behaviors into datums
refactor: Refactored monkey AI a bit, noticeable changes should be minimal!
/:cl:

### CODE REVIEW

File: code/datums/ai/monkey/monkey_subtrees.dm [hunk 1]
```diff
@@ -1,102 +1,63 @@
...
-		if(length(food_candidates))
-			var/obj/item/best_held = controller.GetBestWeapon(null, living_pawn.held_items)
-			for(var/obj/item/held as anything in living_pawn.held_items)
-				if(!held || held == best_held)
-					continue
-				living_pawn.dropItemToGround(held)
+	if(!selected_enemy.stat) //He's up, get him!
+		if(living_pawn.health < MONKEY_FLEE_HEALTH) //Time to skeddadle
+			controller.queue_behavior(/datum/ai_behavior/monkey_flee)
+			return SUBTREE_RETURN_FINISH_PLANNING //I'm running fuck you guys
```
Comment (nednaZ) [inline code/datums/ai/monkey/monkey_subtrees.dm:45]: I'd like a check here to nullify the current target if the target doesn't exist.
I have a reproducible repeating runtime error on line 42 if their target stops existing.

**Review by nednaZ — CHANGES_REQUESTED** [review]
Looks to be fine, though I found one major issue.,
Will testmerge after it's handled.

File: code/datums/ai/monkey/monkey_subtrees.dm (hunk 1 above)
Comment (DoctorSquishy) [reply code/datums/ai/monkey/monkey_subtrees.dm]: I believe I fixed the runtime

**Comment by nednaZ** [comment]
Haven't been seeing any AI issues in the testmerge.

**Pull request merged**